# HG changeset patch
# User Sergi Almacellas Abellana <sergi@koolpi.com>
Take in account access rule on reconcile wizard

issue5512
review18061002

Index: trytond/trytond/modules/account/move.py
===================================================================

--- a/trytond/trytond/modules/account/move.py
+++ b/trytond/trytond/modules/account/move.py
@@ -1835,17 +1835,20 @@
     def get_accounts(self):
         'Return a list of account id to reconcile'
         pool = Pool()
+        Rule = pool.get('ir.rule')
         Line = pool.get('account.move.line')
         line = Line.__table__()
         Account = pool.get('account.account')
         account = Account.__table__()
         cursor = Transaction().cursor
+        rule_expression = Rule.query_get(Account.__name__)

         balance = line.debit - line.credit
         cursor.execute(*line.join(account,
                 condition=line.account == account.id).select(
                 account.id,
-                where=(line.reconciliation == Null) & account.reconcile,
+                where=((line.reconciliation == Null) & account.reconcile
+                    & account.id.in_(rule_expression)),
                 group_by=account.id,
                 having=(
                     Sum(Case((balance > 0, 1), else_=0)) > 0)

